feedback/filters.py

```diff
@@ -156,6 +169,76 @@ def filter(self, qs, value):
         return qs.filter_flags(*value)
 
 
+class ComboTextSearchWidget(forms.MultiWidget):
+    """Widget for displaying a text search field with a checkbox
+    that allows using another search method.
+    The checkbox label and helptext need to be defined through widget
+    attributes 'bool_label' and 'bool_helptext'.
+    """
+    template_name = "feedback/widgets/combo_textsearch_widget.html"
+
+    def __init__(self, attrs=None):
+        widgets = {
+            'text': forms.TextInput(attrs=attrs),
+            'regex': forms.CheckboxInput(attrs=attrs),
+        }
+        super().__init__(widgets, attrs)
+
+    def decompress(self, value) -> tuple[str, bool]:
```

Comment:
Should the parameter `value` have a type annotation?

Reply:
(I also realized that it probably would be a good idea to change this method slightly to check `if value` rather than `if value is None` (and respectively swap the return values) (so if somehow the empty value is something else than None, it still would work).

Reply:
> A very good question. It can either be `None` or a `tuple[str, bool]`

Although I guess it could also be a list consisting of a str and bool, but [list only accepts one type argument](https://docs.python.org/3/library/typing.html#annotating-tuples), so that cannot be provided as a type hint. I guess I just have to stick with the Optional[tuple[str, bool]]